### PR TITLE
[#10569] Missing loading icons and bugs

### PR DESCRIPTION
**Pull request by jfrairigh**
Fixes Part of #10569 
Ready for Review
**Outline of Solution**
This PR has added a loading icon for  all buttons under "Deleted Feedback Sessions" table, which is only the first bullet point of issue #10569.

In instructor-sessions-page.component.ts I added boolean variables for restoring and permanently deleting sessions from recycle bin. I used these variables to track when each process was processing. I did this by setting variables equal to true when the corresponding method was called, and setting back to false when the method had finished the request. In instructor-sessions-page.component.html I added the newly created boolean variables to the *tmIsLoading directive for the session recycle bin table.

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts [hunk 1]
```diff
@@ -458,6 +462,7 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
           this.sessionsTableRowModels.push(m);
           this.updateInstructorPrivilege(m);
           this.statusMessageService.showSuccessToast('The feedback session has been restored.');
+
```
Comment (t-cheepeng) [inline src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts:465]: Unnecessary line change here

**Review by t-cheepeng — COMMENTED** [review]
Thanks for the work. Could you upload a screenshot so we can verify that the intended changes work?

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts (hunk 1 above)
Comment (jfrairigh) [reply src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts]: I have removed the Unnecessary line change. It is difficult to capture the change in a screenshot so I used a video to demo the changes instead.

Here is the link to the video:
(https://drive.google.com/file/d/1Q4soInj7UlcOhUuodINzNfnxtXai4-vO/view?usp=sharing)

**Review by t-cheepeng — APPROVED** [review]
LGTM

**Comment by jfrairigh** [comment]
Is the merge conflict something that I need to resolve?

**Comment by madanalogy** [comment]
> Is the merge conflict something that I need to resolve?

Yeap, please do so and we will follow up on the resolution of this PR. Thanks!

**Review by madanalogy — APPROVED** [review]

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts [hunk 2]
```diff
@@ -574,27 +579,30 @@ export class InstructorSessionsPageComponent extends InstructorSessionModalPageC
...
+        restoredSessions.forEach((session: FeedbackSession) => {
+          this.recycleBinFeedbackSessionRowModels = [];
+          const m: SessionsTableRowModel = {
+            feedbackSession: session,
+            responseRate: '',
+            isLoadingResponseRate: false,
+            instructorPrivilege: session.privileges || DEFAULT_INSTRUCTOR_PRIVILEGE,
+          };
+          this.sessionsTableRowModels.push(m);
+          this.updateInstructorPrivilege(m);
```
Comment (madanalogy) [inline src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts:593]: Hang on, I think this line is no longer necessary, it was removed in #10792

**Review by madanalogy — CHANGES_REQUESTED** [review]

**Comment by jfrairigh** [comment]
I have run into a problem that I am not sure how to resolve. When I pulled code from the Teammates master branch to my local repository "privileges" (ie: feedbackSession.privileges or session.privileges) produces a compiler error of "Property 'privileges' does not exist on type 'FeedbackSession'". I'm sure this is not an issue in Teammates master branch so I'm not sure why it is an issue in my local repository. I'm still getting use to the Git workflow so I'm confident I have done something wrong. Any suggestions would be greatly appreciated.

image

**Comment by madanalogy** [comment]
> I have run into a problem that I am not sure how to resolve. When I pulled code from the Teammates master branch to my local repository "privileges" (ie: feedbackSession.privileges or session.privileges) produces a compiler error of "Property 'privileges' does not exist on type 'FeedbackSession'". I'm sure this is not an issue in Teammates master branch so I'm not sure why it is an issue in my local repository. I'm still getting use to the Git workflow so I'm confident I have done something wrong. Any suggestions would be greatly appreciated.

What you want to do is to make sure you refresh the auto-generated types with `./gradlew generateTypes` or the equivalent if you're on a windows system. Please see the [development guide](https://github.com/TEAMMATES/teammates/blob/master/docs/development.md#managing-the-dev-server-front-end) for more information.

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts (hunk 2 above)
Comment (jfrairigh) [reply src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.ts]: I have fixed the requested changes. I apologize for the numerous commits. I had to fix somethings in my workflow, but everything should be good now.

**Review by madanalogy — APPROVED** [review]
LGTM

**Pull request merged**